miner/miner.go

```diff
@@ -560,6 +565,37 @@ func (m *Miner) mineOne(ctx context.Context, base *MiningBase) (minedBlock *type
 		return nil, err
 	}
 
+	tEquivocateWait := build.Clock.Now()
+
+	// TODO: make param
+	m.niceSleep(time.Until(base.ComputeTime.Add(2 * time.Second)))
+	newBase, err := m.GetBestMiningCandidate(ctx)
+	if err != nil {
+		err = xerrors.Errorf("failed to refresh best mining candidate: %w", err)
+		return nil, err
+	}
+
+	// Only factor in equivocated blocks if doing so will not risk us missing a block
```

Comment:
Needs a comment describing what we're doing (i.e., keeping the intersection between the two tipsets).